client/state/notices/reducer.js

```diff
@@ -31,6 +32,16 @@ export const items = createReducer( {}, {
 
 		return omit( state, noticeId );
 	},
+	[ NOTICE_CLEAR ]: ( state ) => {
+		return reduce( state, ( memo, notice, noticeId ) => {
```

Comment:
Would it be a problem if you were to hard-code an ID into the middleware and reuse that same ID in the form dismiss? One advantage to using a single ID is that it avoids creating multiple of the same notice message (e.g. duplicate "Settings save failed" notices).